### ISSUE TITLE
Why 3D medical segmentation requires so many training epochs？

### ISSUE THREAD

**Issue by lanslotttTT** [MIC-DKFZ/nnUNet#1068]
Thanks for release such a great job！I found that the default training epoch is set to 1000, if I do a 5-fold crossvalidation it will take 5000 training epochs, which seems like it might take me at least a week,and I have a question here, why 3D medical segmentation requires so many training epochs？ Thanks in advance！

**Comment by lukaszbinden** [comment]
Hi, this is really up to you. It doesn't need be a 1000 epochs, you can adjust it. Observe training and validation loss. For instance, I used between 150 and 200 epochs for our publication on a 3D medical segmentation dataset. More than 200 epochs didn't really make much of a difference anymore.

**Comment by us8329** [comment]
> Hi, this is really up to you. It doesn't need be a 1000 epochs, you can adjust it. Observe training and validation loss. For instance, I used between 150 and 200 epochs for our publication on a 3D medical segmentation dataset. More than 200 epochs didn't really make much of a difference anymore.
Hi can you please tell how did you custom set the epochs in nnUNetTrainerV2

**Comment by lanslotttTT** [comment]
> > Hi, this is really up to you. It doesn't need be a 1000 epochs, you can adjust it. Observe training and validation loss. For instance, I used between 150 and 200 epochs for our publication on a 3D medical segmentation dataset. More than 200 epochs didn't really make much of a difference anymore.
> > Hi can you please tell how did you custom set the epochs in nnUNetTrainerV2

You can set it in [network_trainer](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/training/network_training/network_trainer.py),this class is the base class for nnUNetTrainerV2

**Comment by us8329** [comment]
> > > Hi, this is really up to you. It doesn't need be a 1000 epochs, you can adjust it. Observe training and validation loss. For instance, I used between 150 and 200 epochs for our publication on a 3D medical segmentation dataset. More than 200 epochs didn't really make much of a difference anymore.
> > > Hi can you please tell how did you custom set the epochs in nnUNetTrainerV2
> 
> You can set it in [network_trainer](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/training/network_training/network_trainer.py),this class is the base class for nnUNetTrainerV2

Hi  ,thanks for the quick reply , however  I changed the  self.max_num_epochs to 100 , but the training epochs is crossing that , what should i put for self.num_batches_per_epoch and self.num_val_batches_per_epoch

**Comment by FabianIsensee** [comment]
This has been described multiple times already. Please use the search function.

Read this: https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/extending_nnunet.md

In principle, each change requires a separate trainer class. Changing epochs works like this:
```
class nnUNetTrainerV2_2epochs(nnUNetTrainerV2):
    def __init__(self, plans_file, fold, output_folder=None, dataset_directory=None, batch_dice=True, stage=None,
                 unpack_data=True, deterministic=True, fp16=False):
        super().__init__(plans_file, fold, output_folder, dataset_directory, batch_dice, stage, unpack_data,
                         deterministic, fp16)
        self.max_num_epochs = 2
```

If you plan to report the performance of nnU-Net as a baseline, ALWAYS use the default. No tampering with epochs ;-)
Best,
Fabian
